ksql-engine/src/main/java/io/confluent/ksql/function/UdfCompiler.java

```diff
@@ -0,0 +1,158 @@
+   * of the arguments.
+   *<pre>{@code
+   * // try and coerce the arguments to the types expected by the UDF
+   * Boolean arg0;
+   * if(args[0] == null) arg0 = null;
+   * else if(args[0] instanceof Boolean) arg0 = (Boolean)args[0];
+   * else if(args[0] instanceof String) arg0 = Boolean.valueOf((String)args[0]);
+   * else throw new KsqlFunctionException("Type: " + args[0].getClass()
+   * + "is not supported by KSQL UDFS");
+   *
+   * // invoke the udf with the args
+   * return ((UdfCompilerTest) thiz).udf(arg0);
+   *}</pre>
+   * @param method  the UDF to generate the code for
+   * @return String representation of the code that should be compiled for the UDF
+   */
+
+  private static String generateCode(final Method method) {
+    final StringBuilder builder = new StringBuilder();
+    final Class<?>[] params = method.getParameterTypes();
+    for (int i = 0; i < params.length; i++) {
+      final Function<Integer, String> converter = typeConverters.get(params[i]);
+      if (converter == null) {
+        throw new KsqlException("Type " + params[i] + " is not supported in UDFs");
+      }
+      builder.append(converter.apply(i)).append("\n");
+    }
+
+    builder.append("\nreturn ((").append(method.getDeclaringClass().getSimpleName())
+        .append(") thiz).").append(method.getName()).append("(");
+
+    for (int i = 0; i < params.length; i++) {
+      builder.append("arg").append(i).append(",");
+    }
+
+    builder.deleteCharAt(builder.length() - 1);
+    builder.append(");");
+    final String code = builder.toString();
+    logger.debug("generated code for udf method = {}\n{}", method, code);
+    return code;
+  }
+
+  private static IScriptEvaluator createScriptEvaluator(final Method method,
+                                                        final ClassLoader loader) throws Exception {
+    final IScriptEvaluator scriptEvaluator
+        = CompilerFactoryFactory.getDefaultCompilerFactory().newScriptEvaluator();
+    scriptEvaluator.setClassName(method.getDeclaringClass().getName() + "_" + method.getName());
+    final String udfClassName = method.getDeclaringClass().getName();
+    scriptEvaluator.setDefaultImports(new String[]{
+        "java.util.*",
+        "io.confluent.ksql.function.KsqlFunctionException",
+        udfClassName,
+    });
+    scriptEvaluator.setParentClassLoader(loader);
+    return scriptEvaluator;
+  }
+
+  private static String typeConversionCode(final String type, final int index) {
+    if (type.equals("Map") || type.equals("List")) {
+      return type + " arg" + index + " = (" + type + ")args[" + index + "];\n";
+    }
+    final String argArrayVal = "args[" + index + "]";
+    final String argVarAssignment = "arg" + index + " = ";
+    final StringBuilder builder = new StringBuilder();
+    final String numericValue = type.equals("Integer") ? "intValue()" : type.toLowerCase()
+        + "Value()";
+    builder.append(type).append(" arg").append(index).append(";\n")
```

Comment:
I'm with @apurvam  - this is hard to read / maintain. How about a single String, using tokens that you can search an replace?

e.g.

```
final String code =
        "$TYPE$ arg$INDEX$;\n"
        + "if(args[$INDEX$] == null) arg$INDEX$ = null;\n"
        + "else if(args[$INDEX$] instanceof Boolean) arg$INDEX$ = Boolean.valueOf((String)arg$INDEX$)\n";
```